on ppc fixes the errors like below
| ./.libs/libpulsecore-1.1.so: undefined reference to `__sqrt_finite'
| collect2: ld returned 1 exit status

Upstream-Status: Pending

ChangeLog

2012-01-06  Khem Raj  <raj.khem@gmail.com>

	* sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrtf.c: Add __*_finite alias.
	Remove cruft.
	* sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrt.c: Ditto.
	* sysdeps/powerpc/powerpc32/603e/fpu/e_sqrt.c: Ditto.
	* sysdeps/powerpc/powerpc32/603e/fpu/e_sqrtf.c: Ditto.
 
Index: libc/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrt.c
===================================================================
--- libc.orig/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrt.c	2012-01-06 18:07:42.296909187 -0800
+++ libc/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrt.c	2012-01-06 18:09:22.572914856 -0800
@@ -39,14 +39,8 @@
    We find the actual square root and half of its reciprocal
    simultaneously.  */
 
-#ifdef __STDC__
 double
 __ieee754_sqrt (double b)
-#else
-double
-__ieee754_sqrt (b)
-     double b;
-#endif
 {
   if (__builtin_expect (b > 0, 1))
     {
@@ -132,3 +126,4 @@
     }
   return f_wash (b);
 }
+strong_alias (__ieee754_sqrt, __sqrt_finite)
Index: libc/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrtf.c
===================================================================
--- libc.orig/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrtf.c	2012-01-06 18:10:37.068917644 -0800
+++ libc/sysdeps/powerpc/powerpc32/603e/fpu/e_sqrtf.c	2012-01-06 18:11:33.408920635 -0800
@@ -37,14 +37,8 @@
    We find the reciprocal square root and use that to compute the actual
    square root.  */
 
-#ifdef __STDC__
 float
 __ieee754_sqrtf (float b)
-#else
-float
-__ieee754_sqrtf (b)
-     float b;
-#endif
 {
   if (__builtin_expect (b > 0, 1))
     {
@@ -99,3 +93,4 @@
     }
   return f_washf (b);
 }
+strong_alias (__ieee754_sqrtf, __sqrtf_finite)
Index: libc/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrt.c
===================================================================
--- libc.orig/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrt.c	2012-01-06 18:11:51.460925644 -0800
+++ libc/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrt.c	2012-01-06 18:12:39.344924405 -0800
@@ -39,14 +39,8 @@
    We find the actual square root and half of its reciprocal
    simultaneously.  */
 
-#ifdef __STDC__
 double
 __ieee754_sqrt (double b)
-#else
-double
-__ieee754_sqrt (b)
-     double b;
-#endif
 {
   if (__builtin_expect (b > 0, 1))
     {
@@ -132,3 +126,4 @@
     }
   return f_wash (b);
 }
+strong_alias (__ieee754_sqrt, __sqrt_finite)
Index: libc/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrtf.c
===================================================================
--- libc.orig/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrtf.c	2012-01-06 18:13:00.892924586 -0800
+++ libc/sysdeps/powerpc/powerpc64/e5500/fpu/e_sqrtf.c	2012-01-06 18:15:27.992931106 -0800
@@ -37,14 +37,8 @@
    We find the reciprocal square root and use that to compute the actual
    square root.  */
 
-#ifdef __STDC__
 float
 __ieee754_sqrtf (float b)
-#else
-float
-__ieee754_sqrtf (b)
-     float b;
-#endif
 {
   if (__builtin_expect (b > 0, 1))
     {
@@ -99,3 +93,4 @@
     }
   return f_washf (b);
 }
+strong_alias (__ieee754_sqrtf, __sqrtf_finite)
